torch/_inductor/cudagraph_trees.py

```diff
@@ -416,25 +422,41 @@ def cudagraphify(
     constants: Tuple[torch.Tensor, ...] = (),
     placeholders: Tuple[PlaceholderInfo, ...] = (),
     mutated_input_idxs: Tuple[int, ...] = (),
+    compile_id: Optional[CompileId] = None,
 ) -> Tuple[ModelType, OutputType]:
-    manager = get_container(device_index).get_tree_manager()
-    assert not (is_backward and is_inference)
-    mode = (
-        CompilationMode.BACKWARD
-        if is_backward
-        else (CompilationMode.INFERENCE if is_inference else CompilationMode.FORWARD)
-    )
+    # Even though this is technically runtime, we want to capture these overheads for
+    # "compilation metrics" logging.
+    metrics_context = get_metrics_context()
+    with metrics_context, dynamo_timed(
+        "cudagraphify",
+        log_pt2_compile_event=True,
```

Comment:
This should only happen O(1) times per compile, right? Just confirming for retention reasons in PT2 Compile Events.

Reply:
That's right